File: fibonacciNumber509.py
#method = dp
#T=O(n)
#S=O(k) 
class Solution:
    def fib(self, n: int) -> int:
        ans = [0,1]
        for i in range(2,n+1):
            ans.append(ans[i-1]+ ans[i-2]) #find the next fib no.
        return ans[n] #return the nth value from list

File: LeetCode/DP/FibonacciNumber509.java
class Solution {
    public int fib(int n) {
        if(n<=1) return n;
        int[] dp = new int[2];
        int res=0;
        dp[0]=0;
        dp[1]=1;
        for(int i=2;i<=n;i++){
            res=dp[0]+dp[1];
            dp[0]=dp[1];
            dp[1]=res;
        }
        return res;
    }
}
